cirq-pasqal/cirq_pasqal/pasqal_sampler.py

```diff
@@ -100,6 +105,20 @@ def _send_serialized_circuit(
 
         return result
 
+    @cirq._compat.deprecated_parameter(
+        deadline='v0.15',
+        fix='The program.device component is going away.'
+        'Attaching a device to PasqalSampler is now done in __init__.',
+        parameter_desc='program',
+        match=lambda args, kwargs: (
+            len(args) >= 2
+            and isinstance(args[1], cirq.AbstractCircuit)
+            and args[1]._device != cirq.UNCONSTRAINED_DEVICE
```

Comment:
Can _device be None?  I see it set to None above but not to cirq.UNCONSTRAINED_DEVICE.  Same with checks below.

Reply:
The `_device` on a `cirq.Circuit` cannot be None. It can either be `cirq.UUNCONSTRAINED_DEVICE` or a user set one.

The `_device` property on this sampler is an `optional[PasqalDevice]` so it can be none.